tests/simple/testdata/comparisons.textproto

```diff
@@ -204,6 +204,323 @@ section {
       errors: { message: "no such overload" }
     }
   }
+  test {
+    name: "eq_dyn_json_null"
+    expr: "dyn(google.protobuf.Value{}) == null"
+    value: { bool_value: true }
+  }
+  test {
+    name: "not_eq_dyn_bool_null"
+    expr: "dyn(false) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_bytes_null"
+    expr: "dyn(b'') == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_double_null"
+    expr: "dyn(2.1) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_duration_null"
+    expr: "dyn(duration('0s')) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_int_null"
+    expr: "dyn(1) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_list_null"
+    expr: "dyn([]) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_map_null"
+    expr: "dyn({}) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_proto2_msg_null"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "dyn(TestAllTypes{}) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_proto3_msg_null"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "dyn(TestAllTypes{}) == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_string_null"
+    expr: "dyn('') == null"
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_timestamp_null"
+    expr: "dyn(timestamp(0)) == null"
+    value: { bool_value: false }
+  }
+}
+section {
+  name: "eq_wrapper"
```

Comment:
Could you add a bit more detail here or in the proposal about when the wrapper is treated as null (if the wrapper is a member and unset)? I think the distinction here isn't totally intuitive and I'm not sure when this would come up in practice?

Reply:
I added tests to show the difference between unset, empty, and set. Wrapper types are tricky and not always implemented correctly in each of these cases.